core/src/main/java/io/micronaut/core/annotation/AnnotationMetadata.java

```diff
@@ -55,12 +57,14 @@
  * @author Graeme Rocher
  * @since 1.0
  */
-public interface AnnotationMetadata extends AnnotationSource {
+public interface AnnotationMetadata extends AnnotationSource, Memoizer<AnnotationMetadata> {
     /**
      * A constant for representing empty metadata.
      */
     AnnotationMetadata EMPTY_METADATA = new EmptyAnnotationMetadata();
 
+    MemoizerNamespace<AnnotationMetadata> MEMOIZER_NAMESPACE = MemoizerNamespace.create();
```

Comment:
add javadoc

Reply:
Yes it's mutable, that's why it's important to only create a limited number of `MemoizedReference`s.

Reply:
the concern is that users could fiddle with this mutable static fields and create weird bugs